app/code/Magento/CatalogInventory/Model/Plugin/AddStockItemsProducts.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\CatalogInventory\Model\Plugin;
+
+/**
+ * Stock Item as a part of ExtensionAttributes
+ */
+class AddStockItemsProducts
+{
+    /**
+     * @var \Magento\CatalogInventory\Api\StockRegistryInterface
+     */
+    private $stockRegistry;
+
+    /**
+     * @param \Magento\CatalogInventory\Api\StockRegistryInterface $stockRegistry
+     */
+    public function __construct(
+        \Magento\CatalogInventory\Api\StockRegistryInterface $stockRegistry
+    ) {
+        $this->stockRegistry = $stockRegistry;
+    }
+
+    /**
+     * Add stock item information to the product's extension attributes
+     *
+     * @param \Magento\Catalog\Model\ResourceModel\Product\Collection $subject
+     * @param array $result
+     * @return \Magento\Catalog\Model\ResourceModel\Product\Collection
+     */
+    public function afterGetItems(\Magento\Catalog\Model\ResourceModel\Product\Collection $subject, $result)
+    {
+        foreach ($result as $product) {
+            $productExtension = $product->getExtensionAttributes();
```

Comment:
We would have N+1 complexity issue in such case.
I would assume having implementation like in \Magento\CatalogInventory\Observer\AddStockItemsObserver to avoid loading stock item data multiple times.

Reply:
hi @swnsma ,I added observer for "catalog_product_collection_load_after" in cataloginventory module